### PR TITLE
[overlay] Add NSItemProviderReading/Writing methods to UIKit overlay

### PR DESCRIPTION
**Pull request by moiseev**
<rdar://problem/32138581>

### CODE REVIEW

**Comment by moiseev** [comment]
@swift-ci Please test

File: stdlib/public/SDK/UIKit/UIKit.swift [hunk 1]
```diff
@@ -295,3 +295,91 @@ extension UIFocusItem {
...
 }
 #endif
+
+//===----------------------------------------------------------------------===//
+// NSItemProviderReading/Writing support
+//===----------------------------------------------------------------------===//
+
+#if os(iOS)
+
+public protocol _SwiftDragDropItemProvider {
```
Comment (jrose-apple) [inline stdlib/public/SDK/UIKit/UIKit.swift]: I'm a bit leery of adding a whole extra protocol like this. Can we just use _ObjectiveCBridgeable instead?

**Comment by moiseev** [comment]
@jrose-apple Used `_ObjectiveCBridgeable` instead of introducing ad-hoc protocols, but that seems to slow down overlay compilation significantly.

**Comment by moiseev** [comment]
@swift-ci Please test

**Comment by moiseev** [comment]
After a local clean build, I don't see any noticeable regressions in compile time. Merging.

**Pull request merged**